app/assets/javascripts/analytics.js.erb

```diff
@@ -2,112 +2,87 @@
-  $('.lnav').on('click', '.lnav-next-icon', function(e) {
-    _gaq.push(['_trackEvent', 'lesson_navigation', 'click_next_lesson_link', 'lesson_page', 1]);
+//Turbolinks magic happens here...
+document.addEventListener("turbolinks:load", function () {
+
+  $(function() {
+
+    // Fire a custom analytics event whenever any link is clicked
+    $('a').on('click', function(e) {
+      var $target = $(e.target);
+      var href = $target.attr('href');
+
+      // manually set href for elements wrapped in anchor tags
+      // this is because the click event target is actually the
+      // internal element, e.g. an <h3> tag wrapped in <a> tags
+      // will register a click target of <h3> not <a> so the
+      // 'href' property will be undefined and we must check its
+      // parents to acquire the href.
+      if(!href) {
+        href = $target.parents('a').attr('href');
+      }
+
+      // Send separate events for internal and external link clicks
+      var regexp = /^https?:/;
+      if(regexp.test(href)) {
+        _gaq.push(['_trackEvent', 'external_link', 'click', href, 1]);
+      } else {
+        _gaq.push(['_trackEvent', 'internal_link', 'click', href, 1]);
+      }
+    });
+
+    // Fire an event whenever a user marks a lesson completed on the lessons index page
+    // else whenever a student marks a lesson not completed on the lessons index page
+    $('[id^=section-lessons__]').on('click', function(e) {
+      if ($(this).children(":first").data('method') === 'post') {
+        _gaq.push(['_trackEvent', 'lesson_completion', 'mark_completed', $(this).children(":first")[0].href, 1]);
+      } else {
+        _gaq.push(['_trackEvent', 'lesson_completion', 'mark_not_completed', $(this).children(":first")[0].href, 1]);
+      }
+    });
+
+    //see_lesson_buttons.html.erb for lesson button analytics
+
+    // Fire an event when a user clicks on the gitter chat link in the individual lesson page
+    $('#gitter-button').on('click', function(e) {
+      _gaq.push(['_trackEvent', 'gitter_chat', 'click_gitter_chat_link', 'lesson_page', 1]);
+    });
+
+    // Fire an event when a user clicks on the floating chat button
+    $('.chat-floating-btn').click(function(e) {
+      _gaq.push(['_trackEvent', 'gitter_chat', 'click_gitter_chat_floating_button', 'global', 1]);
+    });
+
+    // Fire an event when a user clicks on the gitter chat link in the navbar
+    $('.gitter-chat-link').on('click', function(e) {
+      _gaq.push(['_trackEvent', 'gitter_chat', 'click_gitter_chat_link', 'navbar', 1]);
+    });
+
+    // Fire an event whenever someone tries to sign in the standard way
+    $('#login-button').on('click', function(e) {
+      _gaq.push(['_trackEvent', 'signin_buttons', 'click_normal_signin_button', 'signin_button', 1]);
+    })
+
+    // Fire an event whenever someone tries to sign in with Github by clicking that Github button
+    $('.devise-sign-in').on('click', '.btn-github', function(e) {
```

Comment:
I've been looking for the class `'.devise-sign-in'` but can't seem to find it? does this work?

Reply:
It doesn't work. The sign in and sign up GitHub buttons are identical. I believe if you click either it will sign you up if you don't have an account or sign you in if you do so we can't differentiate between them. We could add analytics for the Google button though?

Reply:
ah your of course right, those forms have the same structure.

I suppose theres a couple of ways we could scope the buttons for sign in and sign up. One way would be by the url path. I think you'd have to do that in javascript.

Looking at the forms in the web inspector 👀 another way you could do it is by the `action` attribute on the form

its `action="/users"` on the sign up form
and `action="/users/sign_in"` on the login form

We'll have to duplicate the tracking for both of the forms but thats fine imo.

Whats your thoughts on this?

Reply:
We can if you want that data. But you can sign up from the login GH button and vice versa so it's only going to be a rough indication of the breakdown between the two...

Reply:
Ohh excellent point. I didn't consider that 🙈 it should hopefully be ok though.